test/controllers/MakePaymentControllerSpec.scala

```diff
@@ -328,4 +328,77 @@ class MakePaymentControllerSpec extends ControllerBaseSpec {
       }
     }
   }
+
+  "Calling the makeFullPaymentHandoff action" when {
+
+    "the user is logged in" should {
+
+      lazy val result = {
+        mockPrincipalAuth()
+        mockAudit()
+        controller.makeFullPaymentHandoff()(fakeRequestWithSession)
+      }
+
+      "return status 303(SEE OTHER)" in {
+        status(result) shouldBe Status.SEE_OTHER
+      }
+      "return the correct redirect location" in {
+        redirectLocation(result) shouldBe Some(mockAppConfig.unauthenticatedPaymentsUrl)
+      }
+    }
+
+    "the user is not logged in" should {
+
+      lazy val result = {
+        mockMissingBearerToken()
+        controller.makeFullPaymentHandoff()(request)
+      }
+
+      "return status 303(SEE OTHER)" in {
+        status(result) shouldBe Status.SEE_OTHER
+      }
+      "return the correct redirect location which should be sign in" in {
+        redirectLocation(result) shouldBe Some(mockAppConfig.signInUrl)
+      }
+    }
+
+    "the user is not authenticated" should {
+
+      lazy val result = {
+        mockInsufficientEnrolments()
+        controller.makeFullPaymentHandoff()(request)
+      }
+
+      "return status 403(Forbidden" in {
+        status(result) shouldBe Status.FORBIDDEN
+      }
+    }
+
+    "the user is insolvent and not continuing to trade" should {
+
+      lazy val result = {
+        mockPrincipalAuth()
+        controller.makeFullPaymentHandoff()(insolventRequest)
+      }
+
+      "return status 403(Forbiddden)" in {
+        status(result) shouldBe Status.FORBIDDEN
+      }
+    }
+
+    "the user is an Agent" should {
+      lazy val result = {
+        mockAgentAuth()
+        mockAudit()
```

Comment:
According to the code there shouldn't be an audit if the user is an agent. Can you remove this mock to confirm. Perhaps Scalamock doesn't flag it if you set a mock up and it isn't called, so it didn't cause an issue